src/Bicep.Core/Semantics/Namespaces/DefaultNamespaceProvider.cs

```diff
@@ -5,29 +5,46 @@
 
 public class DefaultNamespaceProvider : INamespaceProvider
 {
-    private delegate NamespaceType GetNamespaceDelegate(string aliasName, ResourceScope resourceScope, IFeatureProvider features);
+    private delegate NamespaceType GetNamespaceDelegate(string aliasName, ResourceScope resourceScope, IFeatureProvider features, ImportDeclarationSyntax? ids = null);
     private readonly ImmutableDictionary<string, GetNamespaceDelegate> providerLookup;
+    private readonly IAzResourceTypeLoaderFactory azResourceTypeLoaderFactory;
 
-    public DefaultNamespaceProvider(IAzResourceTypeLoader azResourceTypeLoader)
+    public DefaultNamespaceProvider(IAzResourceTypeLoaderFactory azResourceTypeLoaderFactory)
     {
-        var azResourceTypeProvider = new AzResourceTypeProvider(azResourceTypeLoader);
+        this.azResourceTypeLoaderFactory = azResourceTypeLoaderFactory;
         this.providerLookup = new Dictionary<string, GetNamespaceDelegate>
         {
-            [SystemNamespaceType.BuiltInName] = (alias, scope, features) => SystemNamespaceType.Create(alias, features),
-            [AzNamespaceType.BuiltInName] = (alias, scope, features) => AzNamespaceType.Create(alias, scope, azResourceTypeProvider),
-            [K8sNamespaceType.BuiltInName] = (alias, scope, features) => K8sNamespaceType.Create(alias),
+            [SystemNamespaceType.BuiltInName] = (alias, scope, features, ids) => SystemNamespaceType.Create(alias, features),
+            [AzNamespaceType.BuiltInName] = (alias, scope, features, ids) =>
+            {
+                var loader = azResourceTypeLoaderFactory.GetResourceTypeLoader(ids, features);
+                if (loader is null)
+                {
+                    return null!;
+                }
+                string providerVersion = "1.0.0"; //builtin version
+                if (features.DynamicTypeLoading)
+                {
+                    providerVersion = ids?.Specification.Version ?? providerVersion;
+                }
+                return AzNamespaceType.Create(alias, scope, new AzResourceTypeProvider(loader, providerVersion));
+
+            },
+            [K8sNamespaceType.BuiltInName] = (alias, scope, features, ids) => K8sNamespaceType.Create(alias),
         }.ToImmutableDictionary();
     }
 
-    public NamespaceType? TryGetNamespace(string providerName, string aliasName, ResourceScope resourceScope, IFeatureProvider features)
-        => providerLookup.TryGetValue(providerName)?.Invoke(aliasName, resourceScope, features);
+    public NamespaceType? TryGetNamespace(string providerName, string aliasName, ResourceScope resourceScope, IFeatureProvider features, ImportDeclarationSyntax? ids = null)
+    //TODO(asilverman): This is the location where we would like to add support for extensibility providers, we want to add a new key and a new loader for the ext. provider
+        => providerLookup.TryGetValue(providerName)?.Invoke(aliasName, resourceScope, features, ids);
```

Comment:
>ids [](http://example.com/codeflow?start=96&length=3)

nit - You should fully expand the argument name to `importDeclarationSyntax` to match our convention. (The `ids` acronym reads like a collection of identifiers, so that's probably another reason to rename.)